libretro-rs/src/lib.rs

```diff
@@ -49,7 +45,9 @@ pub trait RetroCore {
 
   fn cheat_set(&mut self, env: &RetroEnvironment, index: u32, enabled: bool, code: &str) {}
 
-  fn load_game(&mut self, env: &RetroEnvironment, game: RetroGame) -> RetroLoadGameResult;
+  /// Called when a new instance of the core is needed. The `env` parameter can be used to set-up and/or query values
+  /// required for the core to function properly.
+  fn load_game(env: &RetroEnvironment, game: RetroGame) -> RetroLoadGameResult<Self> where Self: Sized;
```

Comment:
Since this is a required part of the API, I would prefer having the `where Self: Sized` constraint at the trait level.

Reply:
Good point! I'm still fairly new to writing Rust code so that didn't occur to me at the time.

Reply:
> Good point! I'm still fairly new to writing Rust code so that didn't occur to me at the time.

No worries 😄 Happy to help new rust programmers whenever possible.